Realm/RLMManagedArray.mm

```diff
@@ -505,14 +506,31 @@ - (instancetype)freeze {
     }
 
     RLMRealm *frozenRealm = [_realm freeze];
-    auto& parentInfo = _ownerInfo->freeze(frozenRealm);
+    auto& parentInfo = _ownerInfo->resolve(frozenRealm);
     return translateRLMResultsErrors([&] {
         return [[self.class alloc] initWithList:_backingList.freeze(frozenRealm->_realm)
                                      parentInfo:&parentInfo
                                        property:parentInfo.rlmObjectSchema[_key]];
     });
 }
 
+- (instancetype)thaw {
+    if (!self.frozen) {
+        return self;
+    }
+
+    RLMRealmConfiguration *rlmConfig = _realm.configuration;
+    RLMRealm *liveRealm = [RLMRealm realmWithConfiguration:rlmConfig error:nil];
+
+    auto tsr = realm::ThreadSafeReference(_backingList);
```

Comment:
This doesn't need to use realm::ThreadSafeReference; it can just call `_backingList.freeze(liveRealm->_realm)` to do the same thing (and then at that point freeze and thaw can be extracted to a single function which takes the target Realm as a parameter).